projects/packages/connection/tests/php/test-rest-endpoints.php

```diff
@@ -593,6 +593,72 @@ public function test_update_connection_owner_success() {
 		$this->assertEquals( self::$secondary_user_id, Jetpack_Options::get_option( 'master_user' ), 'Connection owner should be updated.' );
 	}
 
+	/**
+	 * Testing the `POST /jetpack/v4/connection` endpoint, aka site disconnect endpoint, on failure.
+	 */
+	public function test_disconnect_site_failures() {
+		// isActive missing.
+		$this->request = new WP_REST_Request( 'POST', '/jetpack/v4/connection' );
+		$this->request->set_header( 'Content-Type', 'application/json' );
+
+		$response      = $this->server->dispatch( $this->request );
+		$response_data = $response->get_data();
+
+		$this->assertSame( 400, $response->get_status() );
+		$this->assertSame( 'Missing parameter(s): isActive', $response_data['message'] );
+
+		// isActive invalid.
```

Comment:
these should be separate unit tests. 1 test 1 flow.

Reply:
Thanks for not letting me be lazy 💯